### PR TITLE
CGAL Tutorial surface reconstruction

### PR DESCRIPTION
**Pull request by sgiraudot**
<s>Small feature: https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/CGAL_tutorial_surface_reconstruction</s>

**Update:** That is a feature https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/CGAL_tutorials

TODO:
- [ ] News Entry

### CODE REVIEW

**Comment by lrineau** [comment]
@sgiraudot The small feature status is "canceled". What about this PR?

**Comment by sloriot** [comment]
It was reclassified as a [feature](https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/CGAL_tutorials).

File: Documentation/doc/Documentation/Doxyfile.in [hunk 1]
```diff
@@ -11,7 +11,8 @@ HTML_HEADER         = ${CGAL_DOC_HEADER}
 LAYOUT_FILE         = ${CGAL_DOC_RESOURCE_DIR}/DoxygenLayout.xml
 GENERATE_TAGFILE    = ${CGAL_DOC_TAG_GEN_DIR}/Manual.tag
 EXAMPLE_PATH        = ${CGAL_Convex_hull_2_EXAMPLE_DIR} \
-                      ${CGAL_Kernel_23_EXAMPLE_DIR}
+                      ${CGAL_Kernel_23_EXAMPLE_DIR} \
+                      ${CMAKE_SOURCE_DIR}/Poisson_surface_reconstruction_3/examples
```
Comment (maxGimeno) [inline Documentation/doc/Documentation/Doxyfile.in]: Errors in doc generation : 
```
warning: source /home/cgal-testsuite/cgal_doc_build/CGAL-5.1-Ic-103/doc/Poisson_surface_reconstruction_3/examples is not a readable file or directory... skipping.
```

**Comment by sgiraudot** [comment]
Error fixed and branch rebased.

**Comment by maxGimeno** [comment]
Warnings in the doc : https://cgal.geometryfactory.com/CGAL/Manual_doxygen_test/CGAL-5.1-Ic-108/logs_master/Documentation.log

File: Documentation/doc/Documentation/Tutorials/Tutorial_reconstruction.txt [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+
+\author Simon Giraudot
+
+Surface reconstruction from point clouds is a core topic in geometry
+processing \cgalCite{cgal:btsag-asosr-16}. It is an ill-posed problem:
+there is an infinite number of surfaces that approximate a single
+point cloud and a point cloud does not define a surface in
+itself. Thus additional assumptions and constraints must be defined by
+the user and reconstruction can be achieved in many different
+ways. This tutorials provides guidances on how to use the different
```
Comment (MaelRL) [inline Documentation/doc/Documentation/Tutorials/Tutorial_reconstruction.txt]:
```suggestion
ways. This tutorials provides guidance on how to use the different
```

File: Documentation/doc/Documentation/Tutorials/Tutorial_reconstruction.txt [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+
+\cgal offers three different algorithms for surface reconstruction:
+
+- \ref Chapter_Poisson_Surface_Reconstruction "Poisson Surface Reconstruction"
+- \ref Chapter_Advancing_Front_Surface_Reconstruction "Advancing Front Surface Reconstruction"
+- \ref Chapter_Scale_space_reconstruction "Scale Space Surface Reconstruction"
+
+
+Because reconstruction is an ill-posed problem, it must be regularized
+via prior knowledge. Differences in prior leads to different
```
Comment (MaelRL) [inline Documentation/doc/Documentation/Tutorials/Tutorial_reconstruction.txt]:
```suggestion
via prior knowledge. Differences in prior lead to different
```

Comment (MaelRL) [reply Documentation/doc/Documentation/Tutorials/Tutorial_reconstruction.txt]: Is "prior" used like this common in the field?

File: Documentation/doc/Documentation/Tutorials/Tutorial_reconstruction.txt [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+possibilities offered by \cgal for dealing with point clouds, for
+surface reconstruction purposes. The following diagram shows an
+overview (not exhaustive) of common reconstruction steps using \cgal
+tools.
+
+\cgalFigureBegin{TutorialsReconstructionFigPipeline, reconstruction.svg}
+Pipeline Overview
+\cgalFigureEnd
+
+We now review some of these steps in more details.
```
Comment (MaelRL) [inline Documentation/doc/Documentation/Tutorials/Tutorial_reconstruction.txt]:
```suggestion
We now review some of these steps in more detail.
```

File: Documentation/doc/Documentation/Tutorials/Tutorial_reconstruction.txt [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+to yield the best results.
+
+Note that this _preprocessing_ step is optional: when the input point
+cloud has no imperfections, reconstruction can be applied to it
+without any preprocessing.
+
+
+\cgalFigureBegin{TutorialsReconstructionFigPreprocessing, reconstruction_preproc.png}
+Comparison of advancing front reconstruction output using different
+preprocessing on the same input. Smooth point cloud was generated
```
Comment (MaelRL) [inline Documentation/doc/Documentation/Tutorials/Tutorial_reconstruction.txt]: 'the' smooth/simplified?

**Comment by sloriot** [comment]
Tested in CGAL-5.1-Ic-114 (Hooray!)

**Comment by sloriot** [comment]
Filter branch history, now 1.1Mo no diff with previous version.

**Comment by sloriot** [comment]
@sgiraudot you can now prepare the news entry

**Pull request merged**